src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -293,13 +293,20 @@ forEachOffer (
 
```

Comment:
I'd move the `auto funds = offers.ownerFunds()` here and do:

```// If the owner is the issuer, the limiting factor is the offer itself
// since they have unlimited funds.
auto const funds = (offer.owner () == offer.issueOut ().account)
    ? ownerGives
    : offers.ownerFunds();```